op-service/testutils/mock_eth_client.go

```diff
@@ -88,43 +88,46 @@ func (m *MockEthClient) ExpectPayloadByNumber(n uint64, payload *eth.ExecutionPa
 }
```

Comment:
returning err pointer in `ExpectPayloadByNumber`.

Reply:
Yeah, this is one of the ones I mean in the description where I say:
> Not all functions were changed: where a test relied on a particular mock implementation detail, I kept the mock the same, rather than changing the test.
Here, `TestBatchQueueOverlappingSpanBatch`'s usage of `PayloadByNumber` means it has to be used this way. Since I wanted to keep the scope of this PR to just changing the `mock`s, and not touching tests, `PayloadByNumber` had to stay how it is.